open_flamingo/train/train.py

```diff
@@ -361,27 +431,14 @@ def main():
             f"After FSDP {torch.cuda.memory_allocated()/1024**3:.3} GB on rank {args.rank}"
         )
 
-    else:
+    elif not args.deepspeed:
         model = model.to(device_id)
         ddp_model = DDP(model, device_ids=[device_id])
 
-    # Initialize gradient checkpointing
-    if args.gradient_checkpointing:
-        non_reentrant_wrapper = functools.partial(
-            checkpoint_wrapper,
-            offload_to_cpu=True,
-            checkpoint_impl=CheckpointImpl.NO_REENTRANT,
-        )
-        apply_activation_checkpointing(
-            ddp_model,
-            checkpoint_wrapper_fn=non_reentrant_wrapper,
-            check_fn=lambda m: getattr(m, "_use_gradient_checkpointing", False)
-            and not isinstance(m, FSDP)
-            and not isinstance(m, CheckpointWrapper),
-        )
-
```

Comment:
Haven't run this to double check, but I think for non-deepspeed, the checkpointing logic needs to come _before_ initializing the optimizer, or the optimizer may be referring to parameters w/o checkpoint wrapper classes, while the model refers to parameters w/ the wrapper. Could we check this?